tools/cmake/toolchains/cbmc.cmake

```diff
@@ -0,0 +1,109 @@
+# gets included separately by each proof.
+# ______________________________________________________________________________
+
+set(CBMC 1)
+
+#  0 no output
+#  1 errors
+#  2 warnings
+#  4 results
+#  6 status information
+#  8 statistical information
+#  9 progress information
+# 10 debug info
+set(CBMC_VERBOSITY 4 CACHE STRING "Verbosity of CBMC and related tools")
+
+# ______________________________________________________________________________
+#     Variables that are not intended to be directly changed by proof-writers
+# ``````````````````````````````````````````````````````````````````````````````
+
+# The list of all proof names (used to check that proof names are unique)
+set(cbmc_proof_names "" CACHE INTERNAL "")
+
+# The cbmc root
+set(cbmc_dir ${CMAKE_SOURCE_DIR}/tools/cbmc)
+
+# The cbmc proof root
+set(cbmc_proofs_dir ${cbmc_dir}/proofs)
+
+# ______________________________________________________________________________
+#     Strings that proof-writers can change for a particular proof in each
+#     proof's CMakeLists.txt
+# ``````````````````````````````````````````````````````````````````````````````
+
+# Argument to cbmc --unwind flag. See also cbmc_unwindset below.
+set(cbmc_unwind 1)
+
+# Setting this to true for a particular proof builds that proof with the
+# --nondet-static flag. See also cbmc_nondet_static_except below.
+set(cbmc_nondet_static OFF)
+
+# Argument to cbmc --object-bits flag
+set(cbmc_object_bits 8)
+
+# Maximum object size that can be modeled by cbmc.
+# Must use +1 until https://github.com/diffblue/cbmc/issues/5096 is resolved.
+set(cbmc_max_object_size "(SIZE_MAX>>(CBMC_OBJECT_BITS+1))")
+
+# ______________________________________________________________________________
+#     Lists that proof-writers can append to in each proof's CMakeLists.txt
+# ``````````````````````````````````````````````````````````````````````````````
+
+# Each CBMC proof has a list of labels; CTest can run all proofs that have a
+# particular label. All CBMC proofs have the 'cbmc' label. Each FreeRTOS module
+# should get its own label, and each individual proof should also get its own
+# label.
+set(cbmc_test_labels "cbmc")
+
+# Argument to cbmc --unwindset flag
+set(cbmc_unwindset)
+
+# Argument to goto-instrument --nondet-static-except flag.
+set(cbmc_nondet_static_except)
+
+set(cbmc_compile_options)
+set(cbmc_compile_definitions)
+set(cbmc_compile_includes)
+set(cbmc_flags)
+
+# ______________________________________________________________________________
+#     Use goto-cc (the CBMC compiler) and do other FreeRTOS-specific setup
+# ``````````````````````````````````````````````````````````````````````````````
+
+include("${CMAKE_CURRENT_LIST_DIR}/find_compiler.cmake")
+
+set(CMAKE_SYSTEM_NAME Generic)
+
+afr_find_compiler(AFR_COMPILER_CC goto-cc)
+afr_find_compiler(AFR_COMPILER_CXX goto-cc)
+set(AFR_COMPILER_ASM "${AFR_COMPILER_CC}" CACHE INTERNAL "")
```

Comment:
Can we skip the paragraph above?  Can't we just set the compilers below to goto-cc?